### PR TITLE
Replace AreLabelsInWhiteList with IsSubset

### LINKED ISSUE #95137: Should the function `AreLabelsInWhiteList` return false when the whitelist is empty?
**What happened**:
The function `AreLabelsInWhiteList` return `false` when whitelist is empty.
The function is implemented as follows:
https://github.com/kubernetes/apimachinery/blob/829ed199f4e0454344a5bc5ef7859a01ef9b8e22/pkg/labels/labels.go#L146
```golang
func AreLabelsInWhiteList(labels, whitelist Set) bool {
	if len(whitelist) == 0 {
		return true
	}

	for k, v := range labels {
		value, ok := whitelist[k]
		if !ok {
			return false
		}
		if value != v {
			return false
		}
	}
	return true
}
```

**What you expected to happen**:
When the whitelist is empty, any key of labels can not in it. Should the function return `false` when whitelist is empty?

### PR DESCRIPTION
**Pull request by 0xff-dev**
Replace function `AreLabelsInWhiteList` with `IsSubset`.
**What type of PR is this?**
/kind cleanup

**Which issue(s) this PR fixes**:

Fixes #95137

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

**Comment by stevenshuang** [comment]
I signed it

**Comment by stevenshuang** [comment]
I signed it

File: plugin/pkg/admission/podnodeselector/admission.go [hunk 1]
```diff
@@ -259,3 +259,20 @@ func (p *Plugin) getNodeSelectorMap(namespace *corev1.Namespace) (labels.Set, er
 	}
 	return selector, nil
 }
+
+func IsSubset(subSet, superSet labels.Set) bool {
```
Comment (deads2k) [inline plugin/pkg/admission/podnodeselector/admission.go:263]: make private please

File: plugin/pkg/admission/podnodeselector/admission.go [hunk 2]
```diff
@@ -148,7 +148,7 @@ func (p *Plugin) Validate(ctx context.Context, a admission.Attributes, o admissi
 	if err != nil {
 		return err
 	}
-	if !labels.AreLabelsInWhiteList(pod.Spec.NodeSelector, whitelist) {
```
Comment (deads2k) [inline plugin/pkg/admission/podnodeselector/admission.go:151]: please remove the original.  I don't think we need it anymore and that's a clear signal to everyone they should re-examine their code.

File: plugin/pkg/admission/podnodeselector/admission.go (hunk 1 above)
Comment (stevenshuang) [reply plugin/pkg/admission/podnodeselector/admission.go:263]: ok, i will modify it.

File: plugin/pkg/admission/podnodeselector/admission.go (hunk 2 above)
Comment (stevenshuang) [reply plugin/pkg/admission/podnodeselector/admission.go:151]: ok. i will remove it.

File: plugin/pkg/admission/podnodeselector/admission.go [hunk 3]
```diff
@@ -259,3 +259,20 @@ func (p *Plugin) getNodeSelectorMap(namespace *corev1.Namespace) (labels.Set, er
 	}
 	return selector, nil
 }
+
+func isSubset(subSet, superSet labels.Set) bool {
+	if len(subSet) == 0 {
```
Comment (deads2k) [inline plugin/pkg/admission/podnodeselector/admission.go:264]: looks like this logic was reversed.  While I understand that the previous behavior was wrong, we cannot change the meaning of the empty superset.

**Comment by deads2k** [comment]
please squash

File: plugin/pkg/admission/podnodeselector/admission.go [hunk 4]
```diff
@@ -259,3 +259,20 @@ func (p *Plugin) getNodeSelectorMap(namespace *corev1.Namespace) (labels.Set, er
 	}
 	return selector, nil
 }
+
+func isSubset(subSet, superSet labels.Set) bool {
+	if len(superSet) == 0 {
+		return false
```
Comment (deads2k) [inline plugin/pkg/admission/podnodeselector/admission.go:265]: Please don't change the logic from the original.  This method isn't for re-use, it's for whatever this admission plugin needs.  Looks like the intent is for an empty list to always allow.

Comment (stevenshuang) [reply plugin/pkg/admission/podnodeselector/admission.go:265]: thanks, I understand it,  the logic should not change. It's my mistake.

**Pull request merged**